<launch>   
    <arg name="object_3d" default="false"/>
    <arg name="gui" default="false"/>
    <arg name="navigation" default="false"/>

    <!-- Start both raspi cameras in a stereo configuration -->
    <!-- Group is to keep remaps from leaking -->
    <group>
        <include file="$(find project_fetch)/launch/camera.launch"/>
    </group>


    <group ns="/stereo">
        <!-- Setup image disparity and rectification processing -->
        <node pkg="stereo_image_proc" type="stereo_image_proc" name="stereo_image_proc" output="screen">
            <param name="queue_size" value="5" />
            <param name="approximate_sync" value="False" />

            <!--
            <remap from="/stereo/right/image_mono" to="/stereo/right/image_raw"/>
            <remap from="/stereo/right/image_raw" to="/stereo/right/async/image_raw"/>
            <remap from="/stereo/left/image_raw" to="/stereo/left/async/image_raw"/>
            <remap from="/stereo/right/camera_info" to="/stereo/right/async/camera_info"/>
            <remap from="/stereo/left/camera_info" to="/stereo/left/async/camera_info"/>
            -->
        </node>
    </group>

    <!-- Start point cloud generation fro navigation -->
    <include if="$(arg navigation)" file="$(find project_fetch)/launch/navigation.launch"/>

    <group if="$(arg object_3d)">
        <include file="$(find project_fetch)/launch/find_object_3d.launch" ns="/stereo">
            <arg name="gui" value="$(arg gui)"/>
        </include>
    </group>
    <group unless="$(arg object_3d)">
        <include file="$(find project_fetch)/launch/find_object.launch" ns="/stereo">
            <arg name="gui" value="$(arg gui)"/>
        </include>
    </group>
</launch>
